### PR TITLE
homely: init at 0.15.3

### PR DESCRIPTION
**Pull request by lokke-me**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/homely/default.nix
Comment (srhb) [inline pkgs/applications/misc/homely/default.nix]: Please format as either

```
{ stdenv
, pythonPackages
}:
```

or

```
{ stdenv, pythonPackages }:
```

File: pkgs/applications/misc/homely/default.nix
Comment (srhb) [inline pkgs/applications/misc/homely/default.nix]: Please indent these using two spaces, like in the rest of the expression.

**Review by srhb — CHANGES_REQUESTED** [review]

**Comment by srhb** [comment]
@GrahamcOfBorg build homely

**Comment by srhb** [comment]
```
----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Looks like there are no tests, so please add `doCheck = false;` as well. :)

**Comment by lokke-me** [comment]
I fixed the mentioned issues, thanks for the feedback

File: pkgs/applications/misc/homely/default.nix
Comment (Infinisil) [inline pkgs/applications/misc/homely/default.nix]: Also works with this removed

**Review by infinisil — CHANGES_REQUESTED** [review]

**Comment by Infinisil** [comment]
Ping? Will merge if you fix above issue

**Comment by Infinisil** [comment]
Oh haha, I just now realized that @srhb said to add `doCheck = false;` and I said to remove it again.

@srhb I think it's better to make it try to run tests anyways, even if there aren't any, because future versions of the package might do have them.

**Comment by srhb** [comment]
@Infinisil I'm only parroting what the actually skilled python maintainers (as opposed to me) tend to say on python PRs, I believe. :)

**Comment by Infinisil** [comment]
Hmm, would like to see the arguments for it, ping @FRidh

**Comment by srhb** [comment]
Think the PR is in a bad state for any sort of discussion right now. Bad rebase?

**Comment by Mic92** [comment]
@srhb asked you to add `doCheck = false;` to the package since we run by default tests for python packages. There are no tests included in the python package. The github release would include tests, but at least some of those require network access: https://github.com/phodge/homely/blob/master/test/test_gitvcs.py

**Comment by Ekleog** [comment]
@lokke-me Sorry for the delay! Could you just rebase your PR on latest master, so that I could trigger a final ofborg run and merge? Feel free to poke me if I don't do in the next few days, or there's a risk that this slips under the radar again :/

File: pkgs/applications/misc/homely/default.nix
Comment (Mic92) [inline pkgs/applications/misc/homely/default.nix:23]:
```suggestion
    homepage = "https://homely.readthedocs.io";
```

File: pkgs/top-level/all-packages.nix
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:17276]:
```suggestion
  homely = python3Packages.callPackage ../applications/misc/homely {};
```

File: pkgs/applications/misc/homely/default.nix
Comment (Mic92) [inline pkgs/applications/misc/homely/default.nix:1]: And than we can list all python modules here explicit instead of pythonPackages.

Comment (lukegb) [reply pkgs/applications/misc/homely/default.nix:1]: As a concrete GitHub suggestion, 

```suggestion
{ stdenv
, buildPythonApplication
, fetchPypi
, attrs
, setuptools_scm
, requests
, simplejson
, click
, pythondaemon
}:

buildPythonApplication rec {
  version = "0.15.3";
  pname = "homely";

  src = fetchPypi {
    inherit pname version;
    sha256 = "2aa7461b583660bf8d9ad776c184fa7a1272141a5c0faaf78a76bc3e8c39db6d";
  };

  buildInputs = [ attrs setuptools_scm ];
  propagatedBuildInputs = [
    requests
    simplejson
    click
    pythondaemon
  ];
```

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 49054` [1](https://github.com/Mic92/nixpkgs-review)

**Comment by lukegb** [comment]
Result of `nixpkgs-review pr 49054` [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/applications/misc/homely/default.nix
Comment (lukegb) [inline pkgs/applications/misc/homely/default.nix:12]: Is `attrs` required? I tried removing it (with Mic92's suggestion) and the package built fine.

**Review by lukegb — CHANGES_REQUESTED** [review]
Builds and runs for me. Upstream has released 0.16 in the meantime (sorry).

As written, this derivation uses Python 2, which is undesirable, but if you take Mic92 and my suggestions then this should be good-to-go.

**Comment by Mic92** [comment]
I think the author lost interest for this one. Maybe someone else could make a PR if they are interested.

**Pull request closed without merging**
